src/lib.rs

```diff
@@ -213,12 +213,26 @@ impl<'object, Endian> DebugInfo<'object, Endian>
             }
 
             // The row we just last iterated to is *after* the address, to the previous row we
-            // saw (stored in current) is the one we want. There better be one...
+            // saw (stored in current) is the one we want. If there is no current, then we have to
+            // give up on locating the address in this unit (and thus in the program too).
+            //
+            // TODO
+            // Can we return partial information here by giving, say, information from the
+            // compilation unit itself?
+            if current.is_none() {
+                return None;
+            }
             let row = current.unwrap();
             let header = rows.header();
 
+            let file = row.file(header);
+            if file.is_none() {
+                // DWARF is invalid -- DebugLine refers to file that does not exist!
+                return None;
```

Comment:
I would generally prefer to define a `enum Error { ... }` and define variants for each of these `return None` cases, eg `Error::ReferenceToUnknownFile`.